sources/AppBundle/Site/Form/RubriqueType.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace AppBundle\Site\Form;
+
+use Afup\Site\Corporate\Feuilles;
+use AppBundle\Site\Model\Rubrique;
+use AppBundle\Association\Model\Repository\UserRepository;
+use AppBundle\Site\Model\Repository\RubriqueRepository;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\Form\Extension\Core\Type\DateType;
+use Symfony\Component\Form\Extension\Core\Type\FileType;
+use Symfony\Component\Form\Extension\Core\Type\IntegerType;
+use Symfony\Component\Form\Extension\Core\Type\TextareaType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class RubriqueType extends AbstractType 
+{
+    private $rubriqueRepository;
+    private $userRepository;
+
+    public function __construct(RubriqueRepository $rubriqueRepository, UserRepository $userRepository)
+    {
+        $this->rubriqueRepository = $rubriqueRepository;
+        $this->userRepository = $userRepository;
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $users = [];
+        foreach ($this->userRepository->getAll() as $user) {
+           $users[$user->getLastName() . ' ' . $user->getFirstName()] = $user->getId();
+        }
+        
+        $feuilles = (new Feuilles($GLOBALS['AFUP_DB']))->obtenirListe('nom, id', 'nom', true);
+        $positions = [];
+        for ($i = 9; $i >= -9; $i--) {
+            $positions[$i] = $i;
+        }
+
+        $rubriques = [];
+        foreach ($this->rubriqueRepository->getAll() as $rubrique) {
+            $rubriques[$rubrique->getNom()] = $rubrique->getId();
+        }
+        $builder
+            ->add('nom', TextType::class, [
+                'label' => 'Nom de la rubrique',
+                'required' => true,
+                'attr' => [
+                    'maxlength' => 255,
+                    'size' => 60
+                ]
```

Comment:
il serait utile de rajouter aussi des constraints en not null quand on a le required à true dans le cas où on ne prends pas en compte le required html (pour avoir aussi la validation coté serveur)

Reply:
D'ac merci, c'est ajouté